ext/ddtrace_profiling_native_extension/extconf.rb

```diff
@@ -128,6 +128,12 @@ def add_compiler_flag(flag)
   $defs << '-DHAVE_PTHREAD_GETCPUCLOCKID'
 end
 
+# On older Rubies, we did not need to include the ractor header (this was built into the MJIT header)
+$defs << '-DNO_RACTOR_HEADER_INCLUDE' if RUBY_VERSION < '3.3'
+
+# On older Rubies, some of the Ractor internal APIs were directly accessible
+$defs << '-DUSE_RACTOR_INTERNAL_APIS_DIRECTLY' if RUBY_VERSION < '3.3'
+
```

Comment:
To be honest, I still wonder a bit if this is the right approach. I've been trying to make the checks "semantic" e.g. have a define that says "this structure is missing" and then using that in the code

```c
#ifdef STRUCTURE_MISSING
... do stuff
```

rather than doing it implicitly using the Ruby version

```c
#ifdef RUBY_2_7+
... do stuff
```

But I'm still not completely convinced on the approach I'm using right now.